Sources/Sentry/SentryTracer.m

```diff
@@ -199,11 +216,135 @@ - (SentryTransaction *)toTransaction
                                             }]];
     }
 
+    if (appStartMeasurement != nil) {
+        [self setStartTimestamp:appStartMeasurement.appStartTimestamp];
+    }
+
     SentryTransaction *transaction = [[SentryTransaction alloc] initWithTrace:self children:spans];
     transaction.transaction = self.name;
+    [self addMeasurements:transaction appStartMeasurement:appStartMeasurement];
     return transaction;
 }
 
+- (SentryAppStartMeasurement *)getAppStartMeasurment
+{
+    SentryAppStartMeasurement *appStartMeasurement = nil;
+
+    // Only send app start measurement for transactions generated by auto performance
+    // instrumentation.
+    if (![self.context.operation isEqualToString:SENTRY_VIEWCONTROLLER_RENDERING_OPERATION]) {
+        return appStartMeasurement;
+    }
+
+    // Double-Checked Locking to avoid acquiring unnecessary locks.
+    if (SentrySDK.appStartMeasurement == nil) {
+        return appStartMeasurement;
+    }
+
+    @synchronized(SentrySDK.appStartMeasurementLock) {
```

Comment:
`appStartMeasurementLock` is a property that insttiates the object we're locking against in a lazily in a non thread-safe way.

The instance you are locking against must be instantiated in an atomic way, visible to all threads.

In other words:

```
if a != null;
 a = object()
```
Isn't atomic so best have `a` have an inlined object initialization

Reply:
Good catch, thanks 💯 .